modules/lwjgl/core/src/main/java/org/lwjgl/Version.java

```diff
@@ -53,4 +55,15 @@ public enum BuildType {
         }
     }
 
+    private static Optional<String> optionalBuildSuffix() {
+        String suffix = Version.class.getPackage().getImplementationVersion();
+        if (suffix != null) return Optional.of(suffix);
+        try {
+          try (JarFile jar = new JarFile(Path.of(Version.class.getProtectionDomain().getCodeSource().getLocation().toURI()).toFile())) {
+            return Optional.ofNullable(jar.getManifest().getMainAttributes().getValue(Attributes.Name.IMPLEMENTATION_VERSION));
+          }
+        } catch (Exception ignored) {
+            return Optional.empty();
+        }
```

Comment:
Instead of loading the JarFile manually, I suggest just loading the `Manifest` manually by retrieving the resource via `Class.getResourceAsStream(String)`. This should work on Java 8 and Java 9+ and even with modules, since "a resource name with the name "META-INF/MANIFEST.MF" is never encapsulated because "META-INF" is not a legal package name." (See the [encapsulation rules for resources](https://docs.oracle.com/en/java/javase/18/docs/api/java.base/java/lang/Module.html#getResourceAsStream(java.lang.String))). A few basic tests seem to confirm that this works as expected.

```suggestion
        try (InputStream in = Version.class.getResourceAsStream("/" + JarFile.MANIFEST_NAME)) {
            if (in == null) return Optional.empty();

            Manifest manifest = new Manifest(in);
            return Optional.ofNullable(manifest.getMainAttributes().getValue(Attributes.Name.IMPLEMENTATION_VERSION));
        } catch (Exception ignored) {
            return Optional.empty();
        }
```

Reply:
Check via:
- `jshell --module-path .lib --add-modules ALL-MODULE-PATH`
- `jshell> org.lwjgl.Version.class.getResource("/" + java.util.jar.JarFile.MANIFEST_NAME)`
- `$1 ==> jar:file:///.../lib/org.lwjgl.jar!/META-INF/MANIFEST.MF`

Reply:
Seems to work and might also spare some memory/OS resources by re-using already loaded bits.